src/io/io.jl

```diff
@@ -1,2 +1,309 @@
 include("write_vtk.jl")
 include("read_vtk.jl")
+
+# handle "_" on optional prefix strings
+add_opt_str_post(str) = (str === "" ? "" : "_$(str)")
+
+function write_meta_data(callback::SolutionSavingCallback, integrator)
+    prefix = callback.prefix
+
+    meta_data = create_meta_data_dict(callback, integrator)
+
+    # write JSON-file
+    output_directory = callback.output_directory
+    mkpath(output_directory)
+    json_file = joinpath(output_directory,
+                         "meta" * add_opt_str_post(prefix) * ".json")
+
+    open(json_file, "w") do file
+        JSON.print(file, meta_data, 2)
+    end
+end
+
+function create_meta_data_dict(callback, integrator)
+    git_hash = callback.git_hash
+    prefix = hasproperty(callback, :prefix) ? callback.prefix : ""
+    semi = integrator.p
+    names = system_names(semi.systems)
+
+    meta_data = Dict{String, Any}()
+
+    info = Dict{String, Any}()
+    add_simulation_info!(info, git_hash, integrator)
+    meta_data["simulation_info"] = info
+
+    systems = Dict{String, Any}()
+    foreach_system(semi) do system
+        idx = system_indices(system, semi)
+        name = prefix * add_opt_str_post("$(names[idx])")
```

Comment:
This is not correct. This function is for optional postfixes, not prefixes, which means you end up with `_fluid_1`.
Use this:
```suggestion
        name = add_opt_str_pre(prefix) * names[idx]
```
with
```julia
add_opt_str_pre(str) = (str === "" ? "" : "$(str)_")
```
It probably makes sense to rename this function to something more descriptive like `add_underscore_to_optional_prefix`.